models/models/billing.py

```diff
@@ -138,6 +78,99 @@ class BillingColumn(str, Enum):
     SEQUENCING_TYPE = 'sequencing_type'
     STAGE = 'stage'
     SEQUENCING_GROUP = 'sequencing_group'
+    COMPUTE_CATEGORY = 'compute_category'
+    CROMWELL_SUB_WORKFLOW_NAME = 'cromwell_sub_workflow_name'
+    CROMWELL_WORKFLOW_ID = 'cromwell_workflow_id'
+    GOOG_PIPELINES_WORKER = 'goog_pipelines_worker'
+    WDL_TASK_NAME = 'wdl_task_name'
+    NAMESPACE = 'namespace'
+
+    @classmethod
+    def can_group_by(cls, value: 'BillingColumn') -> bool:
+        """
+        Return True if column can be grouped by
+        TODO: If any new columns are added above and cannot be in a group by, add them here
+        This could be record, array or struct type
+        """
+        return value not in (
+            BillingColumn.COST,
+            BillingColumn.SERVICE,
+            # BillingColumn.SKU,
+            BillingColumn.PROJECT,
+            BillingColumn.LABELS,
+            BillingColumn.SYSTEM_LABELS,
+            BillingColumn.LOCATION,
+            BillingColumn.USAGE,
+            BillingColumn.CREDITS,
+            BillingColumn.INVOICE,
+            BillingColumn.ADJUSTMENT_INFO,
+        )
+
+    @classmethod
+    def is_extended_column(cls, value: 'BillingColumn') -> bool:
+        """Return True if column is extended"""
+        return value in (
+            BillingColumn.DATASET,
+            BillingColumn.BATCH_ID,
+            BillingColumn.SEQUENCING_TYPE,
+            BillingColumn.STAGE,
+            BillingColumn.SEQUENCING_GROUP,
+            BillingColumn.COMPUTE_CATEGORY,
+            BillingColumn.CROMWELL_SUB_WORKFLOW_NAME,
+            BillingColumn.CROMWELL_WORKFLOW_ID,
+            BillingColumn.GOOG_PIPELINES_WORKER,
+            BillingColumn.WDL_TASK_NAME,
+            BillingColumn.NAMESPACE,
+        )
+
+    @classmethod
+    def str_to_enum(cls, value: str) -> 'BillingColumn':
+        """Convert string to enum"""
+        # all column names have underscore in SQL, but dash in UI / stored data
+        adjusted_value = value.replace('-', '_')
+        str_to_enum = {v.value: v for k, v in BillingColumn.__members__.items()}
+        return str_to_enum[adjusted_value]
+
+    @classmethod
+    def raw_cols(cls) -> list[str]:
+        """Return list of raw column names"""
+        return [
+            'id',
```

Comment:
Should you first use the enum, and then call `.value` for each value, so there's not multiple places for typos?

Reply:
Good idea, replaced all strings with `.value`